src/argilla/server/models/questions.py

```diff
@@ -52,17 +55,41 @@ class RatingQuestionSettingsOption(BaseModel):
     value: int
 
 
-class RatingQuestionSettings(BaseQuestionSettings):
-    type: Literal[QuestionType.rating]
-    options: List[RatingQuestionSettingsOption]
+T = TypeVar("T")
 
+
+class ValidOptionCheckerMixin(BaseQuestionSettings, Generic[T]):
     @property
-    def option_values(self) -> List[int]:
+    def option_values(self) -> List[T]:
         return [option.value for option in self.options]
 
     def check_response(self, response: ResponseValue):
         if response.value not in self.option_values:
             raise ValueError(f"{response.value!r} is not a valid option.\nValid options are: {self.option_values!r}")
 
 
-QuestionSettings = Annotated[Union[TextQuestionSettings, RatingQuestionSettings], Field(..., discriminator="type")]
+class RatingQuestionSettings(ValidOptionCheckerMixin[int], BaseQuestionSettings):
+    type: Literal[QuestionType.rating]
+    options: List[RatingQuestionSettingsOption]
+
+
+class LabelSelectionQuestionSettingsOption(BaseModel):
+    value: str
+    text: str
+    description: Optional[str] = None
+
+
+class LabelSelectionQuestionSettings(ValidOptionCheckerMixin[str], BaseQuestionSettings):
+    type: Literal[QuestionType.label_selection]
+    options: conlist(
+        LabelSelectionQuestionSettingsOption,
+        min_items=LABEL_SELECTION_OPTIONS_MIN_ITEMS,
+        max_items=LABEL_SELECTION_OPTIONS_MAX_ITEMS,
+    )
+    visible_options: Optional[PositiveInt] = None
```

Comment:
I remember why we separate schemas and models for the settings: the field validations make sense only for input schemas models. Maybe we should keep 2 models, for validation/non-validation scenarios.

Reply:
I see, so I will update the PR to add the `pydantic.BaseModel` with the relaxed validation for when reading from the DB.